crates/project_panel/src/project_panel_tests.rs

```diff
@@ -5821,6 +5821,103 @@ async fn test_hide_root(cx: &mut gpui::TestAppContext) {
+    let panel = workspace.update(cx, ProjectPanel::new).unwrap();
+
+    cx.simulate_modifiers_change(gpui::Modifiers {
+        control: true,
+        ..Default::default()
+    });
+
+    select_path_with_mark(&panel, "root/file1.txt", cx);
+    select_path_with_mark(&panel, "root/file2.txt", cx);
+
+    panel.update_in(cx, |panel, window, cx| {
+        panel.compare_selected_files(&CompareSelectedFiles, window, cx);
+    });
+    cx.executor().run_until_parked();
+
+    workspace
+        .update(cx, |workspace, _, cx| {
+            let active_items = workspace
+                .panes()
+                .iter()
+                .filter_map(|pane| pane.read(cx).active_item())
+                .collect::<Vec<_>>();
+            assert_eq!(active_items.len(), 1);
+            active_items
+                .into_iter()
+                .next()
+                .unwrap()
+                .downcast::<FileDiffView>()
+                .expect("Open item should be an FileDiffView");
+        })
+        .unwrap();
+}
+
+#[gpui::test]
+async fn test_compare_files_context_menu(cx: &mut gpui::TestAppContext) {
+    init_test_with_editor(cx);
+
+    let fs = FakeFs::new(cx.executor().clone());
+    fs.insert_tree(
+        "/root",
+        json!({
+            "file1.txt": "content of file1",
+            "file2.txt": "content of file2",
+            "dir1": {}
+        }),
+    )
+    .await;
+
+    let project = Project::test(fs.clone(), ["/root".as_ref()], cx).await;
+    let workspace = cx.add_window(|window, cx| Workspace::test_new(project.clone(), window, cx));
+    let cx = &mut VisualTestContext::from_window(*workspace, cx);
+    let panel = workspace.update(cx, ProjectPanel::new).unwrap();
+
+    // Test 1: When only one file is selected, there should be no compare option
+    select_path(&panel, "root/file1.txt", cx);
+
+    let selected_files_count = panel.update(cx, |panel, cx| panel.collect_selected_files(cx).len());
+    assert_eq!(selected_files_count, 1, "Should only have 1 selected file when selecting a single file");
+
+    // Test 2: When multiple files are selected, there should be a compare option
+    cx.simulate_modifiers_change(gpui::Modifiers {
+        control: true,
+        ..Default::default()
+    });
+    select_path_with_mark(&panel, "root/file1.txt", cx);
+    select_path_with_mark(&panel, "root/file2.txt", cx);
+
+    let selected_files_count = panel.update(cx, |panel, cx| panel.collect_selected_files(cx).len());
+    assert_eq!(selected_files_count, 2, "Should have 2 selected files when multi-selecting");
+
+    // Test case 3: Selecting a directory shouldn't count as a comparable file
+    select_path_with_mark(&panel, "root/dir1", cx);
+
+    let selected_files_count = panel.update(cx, |panel, cx| panel.collect_selected_files(cx).len());
+    assert_eq!(selected_files_count, 2, "Selecting a directory should not affect the number of comparable files");
```

Comment:
Nice, but let's assert the paths exactly here, not the length.
Same in every other place in this test.

Technically, there could be a bug in the impl that leaves selected 2 last items only (very odd claim, sure) and this test won't catch that difference.